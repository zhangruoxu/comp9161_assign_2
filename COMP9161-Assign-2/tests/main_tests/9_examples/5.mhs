main =
let x = 1;
in 
let y = 2;
in 
 let f = (letfun f z = z);
 in 
   if (f (True))
        then (Inl (x + y))
        else (Inr x);

