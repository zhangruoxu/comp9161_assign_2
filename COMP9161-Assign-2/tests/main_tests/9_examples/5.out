main :: (Int + Int) = let x :: Int = 1;
                      in let y :: Int = 2;
                         in let f :: forall a. (a -> a) = (letfun f :: (a -> a) z = z);
                            in if f True
                                  then Inl ((+) x y)
                                  else Inr x;
